test/Microsoft.Dnx.DesignTimeHost.FunctionalTests/Infrastructure/DthTestServer.cs

```diff
@@ -0,0 +1,155 @@
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using Microsoft.Dnx.CommonTestUtils;
+using Microsoft.Dnx.Runtime;
+
+namespace Microsoft.Dnx.DesignTimeHost.FunctionalTests.Infrastructure
+{
+    public class DthTestServer : IDisposable
+    {
+        private const string DthName = "Microsoft.Dnx.DesignTimeHost";
+
+        private readonly string _projectPath;
+        private readonly string _bootstraperExe;
+        private readonly string _dthPath;
+        private readonly List<string> _stdout = new List<string>();
+        private readonly List<string> _stderr = new List<string>();
+        private readonly IDictionary<string, string> _environment;
+
+        private Process _process;
+
+        public DthTestServer(string runtimeHomePath, string projectPath, int port)
+            : this(runtimeHomePath, projectPath, port, new Dictionary<string, string>())
+        {
+        }
+
+        public DthTestServer(string runtimeHomePath, string projectPath, int port, IDictionary<string, string> environment)
+        {
+            _projectPath = projectPath;
+            _bootstraperExe = Path.Combine(runtimeHomePath, "bin", Constants.BootstrapperExeName);
+            _dthPath = Path.Combine(runtimeHomePath, "bin", "lib", DthName, $"{DthName}.dll");
+            _environment = environment;
+
+            Port = port;
+            HostId = Guid.NewGuid().ToString();
+
+            Error = "Unknown";
+        }
+
+        public string HostId { get; }
+
+        public int Port { get; }
+
+        public string Error { get; private set; }
+
+        public void WriteOutput(TextWriter writer)
+        {
+            foreach (var line in _stdout)
+            {
+                writer.WriteLine(line);
+            }
+        }
+
+        public void WriteError(TextWriter writer)
+        {
+            foreach (var line in _stderr)
+            {
+                writer.WriteLine(line);
+            }
+        }
+
+        public bool Start(TimeSpan timeout)
+        {
+            if (!File.Exists(_dthPath))
+            {
+                Error = $"Can't find {DthName} at {_dthPath}.";
+                return false;
+            }
+
+            var processStartInfo = new ProcessStartInfo()
+            {
+                UseShellExecute = false,
+                WorkingDirectory = _projectPath,
+                FileName = _bootstraperExe,
+                Arguments = $"--appbase \"{_projectPath}\" \"{_dthPath}\" {Port} {Process.GetCurrentProcess().Id} {HostId}",
+                RedirectStandardOutput = true,
+                RedirectStandardError = true
+            };
+
+            foreach (var pair in _environment)
+            {
+                processStartInfo.EnvironmentVariables[pair.Key] = pair.Value;
+            }
+
+            processStartInfo.EnvironmentVariables["DNX_TRACE"] = "1";
```

Comment:
Will this fail if the environment variables dictionary doesn't already have the "DNX_TRACE" exist?